massa-ledger/src/ledger_db.rs

```diff
@@ -332,9 +346,12 @@ impl LedgerDB {
         let address_deserializer = AddressDeserializer::new();
         for (key, entry) in ledger {
             let (rest, address) = address_deserializer.deserialize(&key[..]).unwrap();
-            if rest.get(0) == Some(&BALANCE_IDENT) {
+            if rest.first() == Some(&BALANCE_IDENT) {
                 addresses.insert(address, Amount::from_bytes_compact(&entry).unwrap().0);
             }
+            if rest.first() == Some(&BYTECODE_IDENT) && !addresses.contains_key(&address) {
+                addresses.insert(address, Amount::from_str("0").unwrap());
+            }
```

Comment:
That's a good usecase for the [try_insert](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.try_insert) feature.

(require `#![feature(map_try_insert)]`)
```suggestion
            if rest.first() == Some(&BYTECODE_IDENT) {
                addresses.try_insert(address, Amount::from_str("0").unwrap());
            }
```